### DISCUSSION TITLE
mixing dword_t and dword_t*

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#67] [Q&A] [answered]
I don't understand why you are mixing the use of dword_t variable and pointer to dword_t memory zone?
Example:
```
dword_t new_file_mem = file_dialog_open();
```
If the function is returning the memory buffer of the file data, it should be a dword_t* type or even byte_t* type for octet steam.

It is working only because the target memory model is 32bit, but it will break using 64 bit mode.

**Comment by VendelinSlezak** [comment, marked as answer]
This code returns exact value in memory where data starts, so then we can make pointer depending on what size of pointer we need:
```
dword_t *pointer_dword = (dword_t *) (new_file_mem);
word_t *pointer_word = (word_t *) (new_file_mem);
byte_t *pointer_byte = (byte_t *) (new_file_mem);
```
But generally this was mistake from my part from time when I did not understand well that when I have pointer, I can easily get also memory where it points to. So you can find more examples where there is unnecessary variable that contain memory of pointer and then pointer. But optimizing all these take time, so I will do it somewhere along the way.
